scripts/cicd/test_syslog_monitor_memory_usage/run_test.sh

```diff
@@ -0,0 +1,105 @@
+#! /bin/bash
+
+TCP_SERVERS=$1
+UDP_SERVERS=$2
+INTERVAL=$3
+
+echo_with_date() {
+    date +"[%Y-%m-%d %H:%M:%S] $*"
+}
+
+stop_agent() {
+  python scalyr_agent/agent_main.py --config $CONFIG_FILE stop
+}
+
+trap stop_agent EXIT
+
+function generate_config() {
+  TCP_SERVERS=$1
+  UDP_SERVERS=$2
+
+  echo "{" > agent.json
+
+  echo '
+"import_vars": ["SCALYR_API_KEY"],
+"scalyr_server": "agent.scalyr.com",
+"api_key": "$SCALYR_API_KEY",
+"log_rotation_max_bytes": "1073741824",
+"syslog_processing_thread_count": "16",
+
+"server_attributes": {
+   "serverHost": "github-action-memory-test"
+
+},' >> agent.json
+
+echo "\"monitors\": [" >> agent.json
+
+for N in `seq $TCP_SERVERS`;
+do
+  echo "{
+      \"module\":                    \"scalyr_agent.builtin_monitors.syslog_monitor\",
+      \"protocols\":                 \"tcp:$(($N+1600))\",
+      \"accept_remote_connections\": \"true\",
+      \"message_log\": \"tcp_$N.log\"
+    }," >> agent.json
+done
+
+for N in `seq $UDP_SERVERS`;
+do
+  echo "{
+      \"module\":                    \"scalyr_agent.builtin_monitors.syslog_monitor\",
+      \"protocols\":                 \"udp:$(($N+1500))\",
+      \"accept_remote_connections\": \"true\",
+      \"message_log\": \"udp_$N.log\"
+    }," >> agent.json
+done
+
+echo "]" >> agent.json
+
+  echo "}" >> agent.json
+}
+
+generate_config $TCP_SERVERS $UDP_SERVERS
+
+CONFIG_FILE=agent.json
+python scalyr_agent/agent_main.py --config $CONFIG_FILE start
+
+python scalyr_agent/agent_main.py --config $CONFIG_FILE status -v
+
+RATE=2000
+MONITOR_INTERVAL=5
+MEM_ALLOWED=$((32*1024*($UDP_SERVERS+$TCP_SERVERS))) # 32Mb per server
+
+pids=()
+for N in `seq $TCP_SERVERS`;
+do
+	loggen --rate $(($RATE/$TCP_SERVERS)) --interval $INTERVAL --stream --active-connections=5 -Q localhost $(($N+1600)) &
+	pids+=($!)
+done
+
+for N in `seq $UDP_SERVERS`;
+do
+	loggen --rate $(($RATE/$UDP_SERVERS)) --interval $INTERVAL --dgram --active-connections=5 -Q localhost $(($N+1500)) &
+	pids+=($!)
+done
+
+PID=`cat ~/scalyr-agent-dev/log/agent.pid`
+
+while ps -p $pids > /dev/null; do
+  MEM_USED=`cat /proc/$PID/status | grep VmRSS | awk '{print $2}'`
```

Comment:
Can just use `awk '/^VmRSS/ {print $2}' /proc/$PID/status`
I would doublecheck that the units are always in kB, I'm not sure if that's always true

Reply:
Thanks, I will use that.
Luckily it's always in Kb.
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/proc/task_mmu.c?id=39a8804455fb23f09157341d3ba7db6d7ae6ee76#n22